### PR TITLE
Support dimension field id and name for fili 

### PR DESCRIPTION
**Pull request by ayeswarya**
Resolves #

## Description
Support dimension field id and name for fili. Add longName for column parameter name.
## Proposed Changes

-
Screen Shot 2021-10-15 at 12 28 34 PM
Screen Shot 2021-10-15 at 12 28 26 PM
Screen Shot 2021-10-15 at 12 28 15 PM

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: packages/reports/addon/components/dropdown-parameter-picker.ts [hunk 1]
```diff
@@ -16,21 +16,27 @@ interface Args {
...
   }
 
   @action
-  onUpdate(id: unknown) {
-    this.args.onUpdate(this.args.parameterMetadata.id, id);
+  onUpdate(selected: any) {
+    this.args.onUpdate(this.args.parameterMetadata.id, selected.id);
+  }
+
+  get select() {
```
Comment (jkusa) [inline packages/reports/addon/components/dropdown-parameter-picker.ts:37]:
```suggestion
  get selected() {
```

File: packages/reports/addon/components/filter-builders/base.hbs [hunk 2]
```diff
@@ -12,10 +12,10 @@
   <div class="filter-builder" ...attributes>
     <div class="filter-builder__subject xs-col-1-2 sm-col-1-2 col-1-5">
       <div class="filter-builder__subject-name">
-        <span class="name m-r-4">{{this.columnName}}</span>
+        <span class="name m-r-4">{{await this.columnName}}</span>
```
Comment (jkusa) [inline packages/reports/addon/components/filter-builders/base.hbs:15]: do we need `await` here?

File: packages/core/tests/unit/models/report-test.js [hunk 3]
```diff
@@ -146,6 +148,8 @@ module('Unit | Model | report', function (hooks) {
       });
 
       assert.ok(report instanceof DeliverableItem, 'Report should be instance of DeliverableItem');
+      console.log(serialized);
+      console.log(ExpectedReport);
```
Comment (kevinhinterlong) [inline packages/core/tests/unit/models/report-test.js:152]:
```suggestion
```

File: packages/reports/tests/acceptance/report-test.js [hunk 4]
```diff
@@ -758,7 +758,7 @@ module('Acceptance | Navi Report', function (hooks) {
     await click($('.menu-content a:contains("PDF")')[0]);
     let model = await getActualModel(find('.export__download-link').getAttribute('href'));
     assert.equal(
-      model.request.columns.objectAt(2).get('displayName'),
+      await model.request.columns.objectAt(2).get('displayName'),
```
Comment (kevinhinterlong) [inline packages/reports/tests/acceptance/report-test.js:761]: `displayNiceName`

File: packages/data/addon/services/navi-formatter.ts [hunk 5]
```diff
@@ -13,22 +13,51 @@ import type { ResponseRow } from 'navi-data/models/navi-fact-response';
...
 export default class NaviFormatterService extends Service {
+  async formatNiceColumnName(
+    columnMetadata?: ColumnMetadataModel,
+    parameters?: Parameters,
+    alias?: string | null
+  ): Promise<string> {
+    if (alias) {
+      return alias;
+    }
+    const allParams = omit(parameters || {}, 'as');
```
Comment (kevinhinterlong) [inline packages/data/addon/services/navi-formatter.ts:24]: We can remove the omit 'as' from both of these methods

File: packages/data/addon/services/navi-formatter.ts [hunk 6]
```diff
@@ -13,22 +13,51 @@ import type { ResponseRow } from 'navi-data/models/navi-fact-response';
...
+    const paramValues = Object.values(allParams);
+    const parameterMetadata: { id: string; name?: string | undefined; description?: string | undefined }[] = [];
```
Comment (kevinhinterlong) [inline packages/data/addon/services/navi-formatter.ts:26]: Use `ColumnFunctionParametersValues` type?

File: packages/core/addon/models/request/base.ts [hunk 7]
```diff
@@ -83,12 +83,14 @@ export default class Base<T extends ColumnType> extends Fragment.extend(Validati
     set(this, 'parameters', { ...this.parameters, ...parameters });
   }
 
+  //display column name
```
Comment (jkusa) [inline packages/core/addon/models/request/base.ts:86]:
```suggestion
  /**
   * Column display name containing param value ids or alias
   */
    
```

**Review by kevinhinterlong — COMMENTED** [review]
Looks like you might be using an older version of npm because it reverted the lock files to v1 (causing the ~30k lines to be removed)

File: packages/core/addon/models/request/base.ts [hunk 8]
```diff
@@ -83,12 +83,14 @@ export default class Base<T extends ColumnType> extends Fragment.extend(Validati
...
   @computed('alias', 'parameters', 'columnMetadata')
   get displayName(): string {
     const { alias, parameters, columnMetadata } = this;
     return this.naviFormatter.formatColumnName(columnMetadata, parameters, alias);
   }
 
+  //display nice column name formulated by looking up parameter metadata
```
Comment (jkusa) [inline packages/core/addon/models/request/base.ts:93]:
```suggestion
  /**
   * Promise based column display name containing param nice name values or provided alias
   */
```

**Comment by ayeswarya** [comment]
> Looks like you might be using an older version of npm because it reverted the lock files to v1 (causing the ~30k lines to be removed)

What version should I be using.

File: packages/core/addon/models/request/base.ts [hunk 9]
```diff
@@ -39,6 +39,9 @@ export default class Base<T extends ColumnType> extends Fragment.extend(Validati
   @attr('string')
   field!: string;
 
+  @attr('string')
+  alias?: string | null;
```
Comment (kevinhinterlong) [inline packages/core/addon/models/request/base.ts:43]: Will this serialize in filters/sorts (e.g. `filter.alias = 'test'` and `filter.serialize()`)?

We probably can do
```suggestion
  declare alias?: string | null;
```
and then keep `@attr('string')` in `column.ts`

File: packages/core/tests/unit/models/request-test.js [hunk 10]
```diff
@@ -23,13 +23,15 @@ module('Unit | Model | Fragment | BardRequest  - Request', function (hooks) {
               request: {
                 filters: [
                   {
+                    alias: null,
```
Comment (kevinhinterlong) [inline packages/core/tests/unit/models/request-test.js:26]: alias on filter

**Review by kevinhinterlong — COMMENTED** [review]
Just the one file

**Review by kevinhinterlong — APPROVED** [review]

**Review by jkusa — APPROVED** [review]

**Pull request merged**